GCC-5.4 failed to build with both "-Werror" and "./configure --enable-unit"

https://github.com/01org/tpm2-tss/blob/77305c06c39ca165f915657522b3bfd8e21e9b71/test/unit/tcti-socket.c#L102

gcc is complaining when I try to build tss 1.2.0 (tarball tpm2-tss-1.2.0.tar.gz)
```
test/unit/tcti-socket.c: In function ‘__wrap_recv’:
test/unit/tcti-socket.c:101:23: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
     uint8_t *buf_in = (uint8_t*)mock ();
                       ^
cc1: all warnings being treated as errors
```

Found on Ubuntu Core 16 for Raspberry Pi 3, gcc version is 5.4.0:
```
$ gcc --version
gcc (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/arm-linux-gnueabihf/5/lto-wrapper
Target: arm-linux-gnueabihf
Configured with: ../src/configure -v --with-pkgversion='Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-libitm --disable-libquadmath --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-armhf/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-armhf --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-armhf --with-arch-directory=arm --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --enable-multilib --disable-sjlj-exceptions --with-arch=armv7-a --with-fpu=vfpv3-d16 --with-float=hard --with-mode=thumb --disable-werror --enable-multilib --enable-checking=release --build=arm-linux-gnueabihf --host=arm-linux-gnueabihf --target=arm-linux-gnueabihf
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu/Linaro 5.4.0-6ubuntu1~16.04.5) 
```

```
$ wget https://github.com/01org/tpm2-tss/releases/download/1.2.0/tpm2-tss-1.2.0.tar.gz
$ tar xzvf tpm2-tss-1.2.0.tar.gz
$ cd tpm2-tss-1.2.0
$ ./configure --enable-unit && make check
...
...
$ make test/unit/tcti-socket
gcc -DPACKAGE_NAME=\"tpm2-tss\" -DPACKAGE_TARNAME=\"tpm2-tss\" -DPACKAGE_VERSION=\"1.2.0\" -DPACKAGE_STRING=\"tpm2-tss\ 1.2.0\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_URL=\"\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DPACKAGE=\"tpm2-tss\" -DVERSION=\"1.2.0\" -DHAVE_CMOCKA=1 -I.     -I./include -I./sysapi/include -Wall -Werror -Wformat -Wformat-security -fstack-protector-all -fpic -fPIC -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Wno-missing-braces -g -O2 -MT test/unit/test_unit_tcti_socket-tcti-socket.o -MD -MP -MF test/unit/.deps/test_unit_tcti_socket-tcti-socket.Tpo -c -o test/unit/test_unit_tcti_socket-tcti-socket.o `test -f 'test/unit/tcti-socket.c' || echo './'`test/unit/tcti-socket.c
test/unit/tcti-socket.c: In function ‘__wrap_recv’:
test/unit/tcti-socket.c:101:23: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
     uint8_t *buf_in = (uint8_t*)mock ();
                       ^
cc1: all warnings being treated as errors
Makefile:4292: recipe for target 'test/unit/test_unit_tcti_socket-tcti-socket.o' failed
make[1]: *** [test/unit/test_unit_tcti_socket-tcti-socket.o] Error 1
make[1]: Leaving directory '/home/liuqun/tpm2-tss-1.2.0'
Makefile:5101: recipe for target 'check-am' failed
make: *** [check-am] Error 2
```

Same bug also exists in master branch:
```
git clone --depth=1 https://github.com/01org/tpm2-tss --single-branch --branch master
cd tpm2-tss
./bootstrap && ./configure --enable-unit && make check
```
```
$ cd tpm2-tss-2.0.0-dev
$ make check 
# or just:
$ make test/unit/tcti-socket

gcc -DPACKAGE_NAME=\"tpm2-tss\" -DPACKAGE_TARNAME=\"tpm2-tss\" -DPACKAGE_VERSION=\"2.0.0-dev\" -DPACKAGE_STRING=\"tpm2-tss\ 2.0.0-dev\" -DPACKAGE_BUGREPORT=\"https://github.com/01org/tpm2-tss/issues\" -DPACKAGE_URL=\"https://github.com/01org/tpm2-tss\" -DSTDC_HEADERS=1 -DHAVE_SYS_TYPES_H=1 -DHAVE_SYS_STAT_H=1 -DHAVE_STDLIB_H=1 -DHAVE_STRING_H=1 -DHAVE_MEMORY_H=1 -DHAVE_STRINGS_H=1 -DHAVE_INTTYPES_H=1 -DHAVE_STDINT_H=1 -DHAVE_UNISTD_H=1 -DHAVE_DLFCN_H=1 -DLT_OBJDIR=\".libs/\" -DPACKAGE=\"tpm2-tss\" -DVERSION=\"2.0.0-dev\" -DHAVE_CMOCKA=1 -I.     -I./include -I./sysapi/include -I./marshal -Wall -Werror -std=gnu99 -Wformat -Wformat-security -fstack-protector-all -fpic -fPIC -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -Wno-missing-braces -g -O2 -MT test/unit/test_unit_tcti_socket-tcti-socket.o -MD -MP -MF test/unit/.deps/test_unit_tcti_socket-tcti-socket.Tpo -c -o test/unit/test_unit_tcti_socket-tcti-socket.o `test -f 'test/unit/tcti-socket.c' || echo './'`test/unit/tcti-socket.c
test/unit/tcti-socket.c: In function ‘__wrap_recv’:
test/unit/tcti-socket.c:102:23: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
     uint8_t *buf_in = (uint8_t*)mock ();
                       ^
cc1: all warnings being treated as errors
Makefile:3134: recipe for target 'test/unit/test_unit_tcti_socket-tcti-socket.o' failed
make[1]: *** [test/unit/test_unit_tcti_socket-tcti-socket.o] Error 1
make[1]: Leaving directory '/home/liuqun/tpm2-tss-2.0.0-dev'
Makefile:3795: recipe for target 'check-am' failed
make: *** [check-am] Error 2
(classic)liuqun@localhost:~/tpm2-tss-2.0.0-dev$ 

```

--- a/test/unit/tcti-device.c
+++ b/test/unit/tcti-device.c
@@ -128,12 +128,12 @@ tcti_device_log_called_test (void **state)
 ssize_t
 __wrap_read (int fd, void *buffer, size_t count)
 {
-    return (ssize_t)mock ();
+    return mock_type (ssize_t);
 }
 ssize_t
 __wrap_write (int fd, const void *buffer, size_t buffer_size)
 {
-    return (ssize_t)mock ();
+    return mock_type (ssize_t);
 }
 
 typedef struct {

--- a/test/unit/tcti-socket.c
+++ b/test/unit/tcti-socket.c
@@ -85,7 +85,7 @@ __wrap_connect (int                    sockfd,
                 const struct sockaddr *addr,
                 socklen_t              addrlen)
 {
-    return (int)mock ();
+    return mock_type (int);
 }
 /*
  * Wrap the 'recv' system call. The mock queue for this function must have an
@@ -98,8 +98,8 @@ __wrap_recv (int sockfd,
              size_t len,
              int flags)
 {
-    ssize_t  ret = (ssize_t)mock ();
-    uint8_t *buf_in = (uint8_t*)mock ();
+    ssize_t  ret = mock_type (ssize_t);
+    uint8_t *buf_in = mock_ptr_type (uint8_t*);
 
     memcpy (buf, buf_in, ret);
     return ret;
@@ -116,7 +116,7 @@ __wrap_select (int             nfds,
                fd_set         *exceptfds,
                struct timeval *timeout)
 {
-    return (int)mock ();
+    return mock_type (int);
 }
 /*
  * Wrap the 'send' system call. The mock queue for this function must have an
@@ -129,7 +129,7 @@ __wrap_send (int         sockfd,
              int         flags)
 
 {
-    return (TSS2_RC)mock ();
+    return mock_type (TSS2_RC);
 }
 /*
  * This is a utility function used by other tests to setup a TCTI context. It

